src/components/Dropdown/Dropdown.jsx

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import React, { useState, useEffect, createRef } from "react";
+import * as Rivet from "../util/Rivet";
+import {
+  isEscapeKeyPress,
+  isKeyEvent,
+  isRightMouseClick,
+  isTabKeyPress,
+  targets,
+} from "../util/EventUtils.js";
+import { handler, isUnhandledKeyPress } from "./DropdownEventUtils.js";
+import { Button } from "../Button";
+import classNames from "classnames";
+import * as PropTypes from "prop-types";
+import { TestUtils } from "../util/TestUtils";
+
+export const Dropdown = ({
+  toggleDropdownOnClickInside = false,
+  alignRight = false,
+  children,
+  className,
+  label,
+  menuClass,
+  modifier = "secondary",
+  size = "small",
+  ...attrs
+}) => {
+  const align = alignRight ? "right" : "left";
+
+  const handleClickOutside = (event) => {
+    if (event && shouldToggleDropdown(event)) {
+      toggleDropdown(event);
+
+      if (isEscapeKeyPress(event) && toggleButton.current) {
+        // If the user pressed the escape key and we have a current reference to the dropdown toggle button put focus back on it
+        toggleButton.current.focus();
+      }
+    }
+  };
```

Comment:
Can these be moved outside the component declaration so they aren't reinitialized every time the component rerenders? If not, they should go after any `use*` hooks by convention.

Reply:
I moved the hooks at the beginning of the method.